istio-operator/internal/controller/istio/multicluster/remote_secret_propagation_manager.go

```diff
@@ -0,0 +1,155 @@
+package multicluster
+
+import (
+	"context"
+	"fmt"
+
+	kcmv1beta1 "github.com/K0rdent/kcm/api/v1beta1"
+	"github.com/k0rdent/istio/istio-operator/internal/controller/istio"
+	remotesecret "github.com/k0rdent/istio/istio-operator/internal/controller/istio/remote-secret"
+	"github.com/k0rdent/istio/istio-operator/internal/controller/record"
+	"github.com/k0rdent/istio/istio-operator/internal/controller/utils"
+	addoncontrollerv1beta1 "github.com/projectsveltos/addon-controller/api/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type RemoteSecretPropagationManager struct {
+	client client.Client
+}
+
+func New(c client.Client) *RemoteSecretPropagationManager {
+	return &RemoteSecretPropagationManager{
+		client: c,
+	}
+}
+
+func (m *RemoteSecretPropagationManager) TryCreate(ctx context.Context, clusterDeployment *kcmv1beta1.ClusterDeployment) error {
+	log := log.FromContext(ctx)
+	log.Info("Trying to create MultiClusterService for secret propagation")
+
+	exists, err := m.multiClusterServiceExists(ctx, clusterDeployment)
+	if err != nil {
+		return fmt.Errorf("failed to check MultiClusterService existence: %v", err)
+	}
+
+	if exists {
+		log.Info("MultiClusterService already exists")
+		return nil
+	}
+
+	if err := m.createMultiClusterService(ctx, clusterDeployment); err != nil {
+		return fmt.Errorf("failed to create MultiClusterService resource: %v", err)
+	}
+
+	m.sendCreationEvent(clusterDeployment)
+	log.Info("MultiClusterService successfully created")
+	return nil
+}
+
+func (m *RemoteSecretPropagationManager) TryDelete(ctx context.Context, req ctrl.Request) error {
+	log := log.FromContext(ctx)
+	log.Info("Trying to delete MultiClusterService for secret propagation")
```

Comment:
Probably better to move this log before you actually starting to delete the MCS, not before the check if it exists